### PR TITLE
SWATCH-795 Log Http request with payload, URI and total time

### PR DESCRIPTION
**Pull request by kartikshahc**
https://issues.redhat.com/browse/SWATCH-795

**Description**: This logs HTTP request in spring and quarkus with URI path, query parameters, payload and response.

Steps: **For Quarkus swatch-contract**
This won't work until we have org listed in application properties that matches the user org. For testing purpose use FILTER_ORGS=test@example.com,urn:console.redhat.com:service:swatch

**Example 1:**
```
curl -X 'POST' \
  'http://localhost:8000/api/swatch-contracts/internal/rpc/sync/contracts/123' \
  -H 'accept: application/json' \
  -H 'x-rh-swatch-psk: placeholder' \
  -d ''
```

**Response:**
2023-05-07 23:48:55,159 DEBUG [com.red.swa.con.fil.ContractsHTTPFilter] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=1D7DB365-B4E3-4A86-8A1C-54DC0F1DA29E, START_TIME=2023-05-07T23:48:55.159578043-04:00} Http Request UUID 1D7DB365-B4E3-4A86-8A1C-54DC0F1DA29E at start time 2023-05-07T23:48:55.159578043-04:00 with URI /api/swatch-contracts/internal/rpc/sync/contracts/123
2023-05-07 23:48:55,213 WARN  [org.jbo.res.res.i18n] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=1D7DB365-B4E3-4A86-8A1C-54DC0F1DA29E, START_TIME=2023-05-07T23:48:55.159578043-04:00, user=urn:console.redhat.com:service:swatch} RESTEASY002160: Provider instance com.redhat.swatch.clients.rh.partner.gateway.api.resources.ApiExceptionMapper is already registered.  2nd registration is being ignored.
2023-05-07 23:48:55,283 DEBUG [com.red.swa.con.fil.ContractsHTTPFilter] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=1D7DB365-B4E3-4A86-8A1C-54DC0F1DA29E, START_TIME=2023-05-07T23:48:55.159578043-04:00, user=urn:console.redhat.com:service:swatch} Http Request UUID 1D7DB365-B4E3-4A86-8A1C-54DC0F1DA29E took total time 124 milli-seconds
2023-05-07 23:48:55,284 DEBUG [com.red.swa.con.fil.ContractsHTTPFilter] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=1D7DB365-B4E3-4A86-8A1C-54DC0F1DA29E, START_TIME=2023-05-07T23:48:55.159578043-04:00, user=urn:console.redhat.com:service:swatch}  After reading http response class StatusResponse {
    status: FAILED
    message: 123 not found in table
}

**Example 2:**
```
curl -X 'GET' \
  'http://localhost:8000/api/swatch-contracts/internal/contracts?org_id=org123&product_id=BASILISK&metric_id=Instance-hours&vendor_product_code=6n58d3s3qpvk22dgew2gal7w3&billing_provider=aws&billing_account_id=123456789101&timestamp=2022-03-10T12%3A15%3A50-04%3A00' \
  -H 'accept: application/json' \
  -H 'x-rh-swatch-psk: placeholder'
```

**Response:**
2023-05-07 23:53:12,664 DEBUG [com.red.swa.con.fil.ContractsHTTPFilter] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=C128DA67-B561-4A0D-A054-8306D4F58C7D, START_TIME=2023-05-07T23:53:12.663840534-04:00} Http Request UUID C128DA67-B561-4A0D-A054-8306D4F58C7D at start time 2023-05-07T23:53:12.663840534-04:00 with URI /api/swatch-contracts/internal/contracts
2023-05-07 23:53:12,666 DEBUG [com.red.swa.con.fil.ContractsHTTPFilter] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=C128DA67-B561-4A0D-A054-8306D4F58C7D, START_TIME=2023-05-07T23:53:12.663840534-04:00} Query parameters metric_id:[Instance-hours] , billing_provider:[aws] , vendor_product_code:[6n58d3s3qpvk22dgew2gal7w3] , org_id:[org123] , timestamp:[2022-03-10T12:15:50-04:00] , product_id:[BASILISK] , billing_account_id:[123456789101] 
2023-05-07 23:53:12,735 WARN  [org.jbo.res.res.i18n] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=C128DA67-B561-4A0D-A054-8306D4F58C7D, START_TIME=2023-05-07T23:53:12.663840534-04:00, user=urn:console.redhat.com:service:swatch} RESTEASY002160: Provider instance com.redhat.swatch.clients.rh.partner.gateway.api.resources.ApiExceptionMapper is already registered.  2nd registration is being ignored.
2023-05-07 23:53:13,071 DEBUG [com.red.swa.con.fil.ContractsHTTPFilter] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=C128DA67-B561-4A0D-A054-8306D4F58C7D, START_TIME=2023-05-07T23:53:12.663840534-04:00, user=urn:console.redhat.com:service:swatch} Http Request UUID C128DA67-B561-4A0D-A054-8306D4F58C7D took total time 407 milli-seconds
2023-05-07 23:53:13,071 DEBUG [com.red.swa.con.fil.ContractsHTTPFilter] (executor-thread-0) {HTTP_REQUEST_MDC_UUID_KEY=C128DA67-B561-4A0D-A054-8306D4F58C7D, START_TIME=2023-05-07T23:53:12.663840534-04:00, user=urn:console.redhat.com:service:swatch}  After reading http response []

**Steps: For spring swatch-core**

This won't work until we have org listed in application yaml that matches the user org. For testing purpose use FILTER_ORGS=org123 and set resteasy: DEBUG in yaml file

Execute:
`USER_USE_STUB=true SUBSCRIPTION_USE_STUB=true CG_USE_STUB=true CLOUDIGRADE_ENABLED=false DEV_MODE=true ./gradlew clean :bootRun`

**Example 3:**

```
curl -X 'GET' \
  'http://localhost:8000/api/rhsm-subscriptions/v1/tally/products/OpenShift%20Container%20Platform/Cores?granularity=Hourly&beginning=2017-07-21T17%3A32%3A28Z&ending=2017-07-21T17%3A32%3A28Z&use_running_totals_format=false' \
  -H 'accept: application/vnd.api+json' \
  -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='
```

**Response:**
2023-05-08 04:09:57,708 [thread=http-nio-8000-exec-3] [DEBUG] [org.candlepin.subscriptions.resteasy.ResteasyHTTPFilter] org123- Http Request UUID 8152A372-E182-4994-BF2E-A0DC830CCFB3 at start time 2023-05-08T04:09:57.708230814Z with URI /tally/products/OpenShift Container Platform/Cores
2023-05-08 04:09:57,709 [thread=http-nio-8000-exec-3] [DEBUG] [org.candlepin.subscriptions.resteasy.ResteasyHTTPFilter] org123- Query parameters granularity:[Hourly], beginning:[2017-07-21T17:32:28Z], ending:[2017-07-21T17:32:28Z], use_running_totals_format:[false]
2023-05-08 04:09:57,750 [thread=http-nio-8000-exec-3] [DEBUG] [org.candlepin.subscriptions.resteasy.ResteasyHTTPFilter] org123- Http Request UUID 8152A372-E182-4994-BF2E-A0DC830CCFB3 took total time 42 milli-seconds
2023-05-08 04:09:57,750 [thread=http-nio-8000-exec-3] [DEBUG] [org.candlepin.subscriptions.resteasy.ResteasyHTTPFilter] org123-  After reading http response class Errors {
    errors: [class Error {
        status: 400
        code: SUBSCRIPTIONS1001
        title: OpenShift Container Platform does not support any granularity finer than Hourly
        detail: null
    }]
}

### CODE REVIEW

**Comment by kartikshahc** [comment]
Difference between Filters and Interceptor in this PR is quite minimum.
image

**Comment by kartikshahc** [comment]
Didn't find quarkus equivalent of logbook in https://mvnrepository.com/artifact/org.zalando so skipped this approach

**Comment by kartikshahc** [comment]
Tried access logging but doesn't support logging request body. It does support all other features below is sample response for code.

2023-04-26 14:26:11,663 INFO  [io.qua.htt.access-log] (executor-thread-0) 
POST /api/swatch-contracts/internal/contracts HTTP/1.1
Host: localhost:8000
User-Agent: curl/7.61.1
accept: application/json
x-rh-swatch-psk: placeholder
Content-Type: application/json
Content-Length: 292

File: swatch-contracts/src/main/java/com/redhat/swatch/contract/filters/ContractsHTTPFilter.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  @ConfigProperty(name = "FILTER_ORGS")
+  Optional<List<String>> filterOrgs;
+
+  public static final String HTTP_REQUEST_MDC_UUID_KEY = "HTTP_REQUEST_MDC_UUID_KEY";
+  public static final String START_TIME = "START_TIME";
+
+  @Context SecurityContext securityContext;
+
+  @Override
+  @SuppressWarnings("Duplicates")
```
Comment (kflahert) [inline swatch-contracts/src/main/java/com/redhat/swatch/contract/filters/ContractsHTTPFilter.java:59]: We can remove this @SupressWarnings as we discussed marking issue as "Not going to fix" in sonar.

**Pull request merged**